api/src/main/java/org/openmrs/module/queue/utils/QueueUtils.java

```diff
@@ -74,4 +74,28 @@ public static double computeAverageWaitTimeInMinutes(List<QueueEntry> queueEntri
 		}
 		return averageWaitTime;
 	}
+	
+	/**
+	 * @param startDate1, endDate1 - the start and end date of one timeframe
+	 * @param startDate2, endDate2 - the start and end date of second timeframe
+	 * @return boolean - indicating whether the timeframes overlap
+	 */
+	public static boolean datesOverlap(Date startDate1, Date endDate1, Date startDate2, Date endDate2) {
```

Comment:
@cioan - I rewrote this implementation, as there were several use cases that it didn't seem to cover.  Also, I updated the handling of nulls so that they indicate an unbounded condition (eg. overlapping nulls should be overlapping, not non-overlapping).  Have a look at this new implementation and let me know if you see any issues.